src/sdk/namespace_fs.js

```diff
@@ -1504,14 +1512,27 @@ class NamespaceFS {
                         // remove a version (or delete marker) with null version ID from .versions/ (if exists)
                         await this._delete_null_version_from_versions_directory(key, fs_context);
                     }
+                    if (is_disabled_dir_content) {
+                        await this._clear_user_xattr(fs_context, path.dirname(latest_ver_path), XATTR_USER_PREFIX);
+                    }
                 }
-                if (latest_ver_info &&
-                    ((this._is_versioning_enabled()) ||
-                        (this._is_versioning_suspended() && latest_ver_info.version_id_str !== NULL_VERSION_ID))) {
-                    dbg.log1('NamespaceFS._move_to_dest_version version ID of the latest version is a unique ID - the file will be moved it to .versions/ directory');
-                    await native_fs_utils._make_path_dirs(versioned_path, fs_context);
-                    await native_fs_utils.safe_move(fs_context, latest_ver_path, versioned_path, latest_ver_info,
+                if (((this._is_versioning_enabled()) ||
+                        (this._is_versioning_suspended() && latest_ver_info && latest_ver_info.version_id_str !== NULL_VERSION_ID))) {
+                    if (latest_ver_info) {
+                        dbg.log1('NamespaceFS._move_to_dest_version version ID of the latest version is a unique ID - the file will be moved it to .versions/ directory');
+                        await native_fs_utils._make_path_dirs(versioned_path, fs_context);
+                        await native_fs_utils.safe_move(fs_context, latest_ver_path, versioned_path, latest_ver_info,
                         gpfs_options?.move_to_versions, bucket_tmp_dir_path);
+                    }
+                    if (is_disabled_dir_content) {
+                        dbg.log1('NamespaceFS._move_to_dest_version latest object is a directory object with attributes on the directory. move the xattr to the new .version file');
+                        //this scenario happens only after moving from disabled to enabled mode or after upgrade. version-id is always null
```

Comment:
let's move it to a function and explain exactly what is done here

Reply:
in general if you choose keeping xattr implementation in content flow, let's try to organize it better

Reply:
I moved all handling of the xattr to a single function, separate from the original flow, since the original flow is the same for directory objects. it can remain the same. its good, because this will also be relevant for delete latest operation